tests/test_precompiled.py

```diff
@@ -512,15 +513,15 @@ def test_is_notify_tag_calls_extract_with_wider_numbers(mocker):
     )
 
 
-def test_rewrite_address_block_end_to_end(mocker):
-    mock_escape = mocker.patch(
-        "app.precompiled.escape_special_characters_for_regex",
-        return_value='MR J DOE\n13 TEST LANE\nTESTINGTON\nTE57 1NG'
-    )
-    new_pdf, address, message = rewrite_address_block(BytesIO(example_dwp_pdf))
+@pytest.mark.parametrize(['pdf_data', 'address_snippet'], [
```

Comment:
Is the change to this test related to the change to the code above? Just can't quite put the two together

Reply:
so this is taking the existing test, except parametrised to check both `example_dwp_pdf` and `valid_letter` - because `valid_letter` has mac fonts, while example_dwp_pdf has windows fonts. The valid_letter test case failed in my initial iteration of the code.

then i tweaked the assertions to make sure it returns the address too (previously it assigned the `address` variable but didn't check it.
i deleted the `escape_special_characters_for_regex` mock because it didn't feel appropriate for a test that specifically refers to being an end_to_end test.

Reply:
Ok, cheers. So it's not technically testing the new lines of code you added, it's more improving and adding coverage for an existing test, right?

If that's the case, is there any way to test the new functionality you've added regarding fonts with differences? If it's a pain to test you could maybe at the least add a simple unit test for `getEncoding` rather than trying to give `toUnicode` a PDF using a weird font